src/pages/o/[orgId]/campaigns/[campId]/events.tsx

```diff
@@ -1,65 +1,92 @@
 import { GetServerSideProps } from 'next';
+import { dehydrate } from 'react-query/hydration';
+import { QueryClient, useQuery } from 'react-query';
 
-export const getServerSideProps : GetServerSideProps = async (context) => {
-    let props;
+function getCampaignEvents(orgId, campId) {
+    return async () => {
+        try {
+            const eventsRes = await fetch(`http://localhost:3000/api/orgs/${orgId}/campaigns/${campId}/actions`);
+            const eventsData = await eventsRes.json();
 
-    try {
-        const { orgId, campId } = context.params;
+            return eventsData.data;
+        }
+        catch (err) {
+            if (err.name != 'FetchError') {
+                throw err;
+            }
+            return null;
+        }
+    };
+}
 
-        const eventsRes = await fetch(`http://api.zetk.in/v1/orgs/${orgId}/campaigns/${campId}/actions`);
-        const eventsData = await eventsRes.json();
-        const cIdRes = await fetch(`http://api.zetk.in/v1/orgs/${orgId}/campaigns/${campId}`);
-        const cIdData = await cIdRes.json();
-        const oRes = await fetch(`https://api.zetk.in/v1/orgs/${orgId}`);
-        const oData = await oRes.json();
+function getCampaign(orgId, campId) {
+    return async () => {
+        try {
+            const cIdRes = await fetch(`http://localhost:3000/api/orgs/${orgId}/campaigns/${campId}`);
+            const cIdData = await cIdRes.json();
 
-        props = {
-            campaign: cIdData.data,
-            events: eventsData.data,
-            org: oData.data,
-        };
-    }
-    catch (err) {
-        if (err.name != 'FetchError') {
-            throw err;
+            return cIdData.data;
         }
-    }
+        catch (err) {
+            if (err.name != 'FetchError') {
+                throw err;
+            }
+            return null;
+        }
+    };
+}
 
-    if (props) {
-        return { props };
-    }
-    else {
-        return {
-            notFound: true,
-        };
-    }
-};
+function getOrg(orgId) {
+    return async () => {
+        try {
+            const oRes = await fetch(`http://localhost:3000/api/orgs/${orgId}`);
+            const oData = await oRes.json();
+
+            return oData.data;
+        }
+        catch (err) {
+            if (err.name != 'FetchError') {
+                throw err;
+            }
+            return null;
+        }
+    };
+}
+
+export const getServerSideProps : GetServerSideProps = async (context) => {
+    const queryClient = new QueryClient();
+    const { orgId, campId } = context.params;
+
+    await queryClient.prefetchQuery('campaignEvents', getCampaignEvents(orgId, campId));
```

Comment:
The key `campaignEvents` will be the same regardless of campaign, which is not great. Browsing between two different campaigns will reuse the events from cache, even though they should be different.